### ISSUE TITLE
When MenuItem has multi-level submenus, the addMenuElementLinks function will execute incorrectly

### ISSUE THREAD

**Issue by zzhblh** [infor-design/enterprise#6120]
**Describe the bug**
When MenuItem has multi-level submenus, the addMenuElementLinks function will execute incorrectly

**To Reproduce**
1. go to https://design.infor.com/code/ids-enterprise/latest/demo/components/toolbar-flex/example-more-actions-predefined.
2. add `console.log('item '+ i + ' : ' + elems[i].innerText + ' linked');` in sohoxi.js (line:47804).
![image](https://user-images.githubusercontent.com/8154831/153542640-4f6af599-2b70-4aec-95a5-2549f2dfe179.png)
3. then run `$('.flex-toolbar').data('toolbar-flex').toPopupmenuData()` in console.

**Expected behavior**
![image](https://user-images.githubusercontent.com/8154831/153542779-d64730ba-9dbb-46c2-9828-b2ca7235cb24.png)

You can see that item 4 is not linked,the recursion is incorrect. Only child elements of the first level should be selected each time you recurse, so `var elems = menu.querySelectorAll(':scope > li:not(.heading)');` should be `var elems = menu.querySelectorAll(':scope > li:not(.heading)');`  and `submenu.querySelectorAll(':scope > li:not(.heading)');` should be `submenu.querySelectorAll(':scope > li:not(.heading)');` 
**Version**
v4.56.0

**Screenshots**
![image](https://user-images.githubusercontent.com/8154831/153542640-4f6af599-2b70-4aec-95a5-2549f2dfe179.png)
![image](https://user-images.githubusercontent.com/8154831/153542779-d64730ba-9dbb-46c2-9828-b2ca7235cb24.png)
**Platform**
All

**Additional context**

**Comment by tmcconechy** [comment]
@zzhblh i dont 100% understand this issue. Whats the actual problem here? Maybe this just needs to be explained another way as hesitant to touch this function when it seems like everything is working properly.

By properly i mean:
1. go to https://design.infor.com/code/ids-enterprise/latest/demo/components/toolbar-flex/example-more-actions-predefined
2. in the menu button pick and item and watch the toast message has the correct data...

**Comment by zzhblh** [comment]
If you have a more complex menu and execute `$('.flex-toolbar').data('toolbar-flex').toPopupmenuData()`, an error will be reported. Because recursion is wrong.
![image](https://user-images.githubusercontent.com/8154831/153788408-7918cb36-8f66-48a1-8a0a-6582fb021ae1.png)

![image](https://user-images.githubusercontent.com/8154831/153788238-ca6cf90d-9295-4553-9fd1-bf140a649e69.png)

**Comment by tmcconechy** [comment]
@ericangeles are you able to follow this enough to work on?

**Comment by ericangeles** [comment]
@zzhblh do you have a case scenario where you encountered this issue?

**Comment by ericangeles** [comment]
Also, I don't see any differences in the `Expected Behavior` you mentioned above.

You can see that item 4 is not linked,the recursion is incorrect. Only child elements of the first level should be selected each time you recurse, so `var elems = menu.querySelectorAll(':scope > li:not(.heading)');` should be `var elems = menu.querySelectorAll(':scope > li:not(.heading)');` and `submenu.querySelectorAll(':scope > li:not(.heading)');` should be `submenu.querySelectorAll(':scope > li:not(.heading)');`

**Comment by zzhblh** [comment]
@ericangeles 
go to https://design.infor.com/code/ids-enterprise/latest/demo/components/toolbar-flex/example-more-actions-predefined.
Menu Button Sub-Item One linked twice.  Menu Button item 4 not linked.
![image](https://user-images.githubusercontent.com/8154831/162115248-769a1725-8d57-4fed-b337-fcb33aae15dc.png)
![image](https://user-images.githubusercontent.com/8154831/162114119-6aee2b19-e609-4c7f-af1a-ceaab2c8409a.png)
====================After change to ':scope > li:not(.heading)'===============================



Menu Button item 4 linked.
![image](https://user-images.githubusercontent.com/8154831/162113692-31a84543-6ccb-4bcb-baa1-ff48a25af947.png) 
![image](https://user-images.githubusercontent.com/8154831/162114563-04ba0ec8-8b7d-4135-8988-55c0c0c3dbde.png)

**Comment by rvaja-infor** [comment]
@tmcconechy / @ericangeles , would there be any chance of a hotfix to the resolution of this in 4.61.x, please?

**Comment by rvaja-infor** [comment]
@tmcconechy the hotfix would be ideal as we are approaching release, and would rather not destabilize the release at this point.
Let me know if it would be possible, thanks.

**Comment by tmcconechy** [comment]
yes will patch a. 4.61.2 -> in a few hours or latest tomorrow (waiting for. someone to answer my question on the release stuff)

**Comment by tmcconechy** [comment]
@rvaja-infor - done its: https://github.com/infor-design/enterprise/blob/4.61.2
Be aware you should update to now earlier than 4.64.0 - next time you do update as i not patching all in between releases

**Comment by rvaja-infor** [comment]
Thank you for that @tmcconechy. Normally we would but management don't want to derail the stability at this time.

**Comment by jbrcna** [comment]
QA Passed. v4.64.0-dev
<img width="453" alt="image" src="https://user-images.githubusercontent.com/56722184/169751086-316138b3-898d-4b6a-95e2-8c3012224aa2.png">

`$('.flex-toolbar').data('toolbar-flex').toPopupmenuData()`
<img width="416" alt="image" src="https://user-images.githubusercontent.com/56722184/169751169-ce165989-b1b5-439e-8781-97f823352a85.png">
